features/steps/steps/attributes.py

```diff
@@ -0,0 +1,16 @@
+from validation_handling import validate_step, StepResult
+from utils import ifc
+from behave import *
+
+
+@validate_step('The {representation_id} shape representation has RepresentationType "{representation_type}"')
+def step_impl(context, **kwargs):
+    inst = kwargs.get('inst', None)
+    representation_id = kwargs.get('representation_id', None)
+    representation_type = kwargs.get('representation_type', None)
```

Comment:
The idea was to account for both `given` and `then` statements. If it's a `then` statement, we iterate over instances and we will have three arguments: `inst`, `representation_id` and `representation_type`. In case of 'Given', it's only the latter two arguments. 
It's applied in the rule GEM002
```
    Given An IfcSpace

    Then There must be one Body shape representation
    Then The Body shape representation has RepresentationType "SweptSolid, Clipping, Brep"
    
    Scenario 2
    
    Given An IfcSpace
    Given The Body shape representation has RepresentationType "Brep" 

    Then There must be one FootPrint shape representation
```

Reply:
Ah ok, I see, then it's related to the comments below https://github.com/buildingSMART/ifc-gherkin-rules/pull/119/files/4961cfa7e08fe5693dd6e0600e1f0d37e82a2e7c#r1435550346 https://github.com/buildingSMART/ifc-gherkin-rules/pull/119/files/4961cfa7e08fe5693dd6e0600e1f0d37e82a2e7c#r1435550634

I think in the case of a given `context.instances = filter(..., context.instances)` should also be applied in the decorator. Just filter based on whether we're getting outcomes of a certain severity for the instance and do the loop (`list(filter(...))`) is also a loop in the decorator.